Xamarin.Forms.Platform.WinRT/Platform.cs

```diff
@@ -761,8 +761,35 @@ void OnPageActionSheet(Page sender, ActionSheetArguments options)
 				dialog.CancelCommandIndex = (uint)dialog.Commands.Count - 1;
 			}
 
-			IUICommand command = await dialog.ShowAsync();
+			IUICommand command = await dialog.ShowAsyncQueue();
 			options.SetResult(command.Label == options.Accept);
 		}
 	}
+	
+	// refer to http://stackoverflow.com/questions/29209954/multiple-messagedialog-app-crash for why this is used
+	// in order to allow for multiple MessageDialogs, or a crash occurs otherwise
+	public static class MessageDialogExtensions
+	{
+		static TaskCompletionSource<MessageDialog> _currentDialogShowRequest;
+
+		public static async Task<IUICommand> ShowAsyncQueue(this MessageDialog dialog)
+		{
+			if (!Window.Current.Dispatcher.HasThreadAccess)
+			{
+				throw new InvalidOperationException("This method can only be invoked from UI thread.");
```

Comment:
Instead of throwing an exception here, does it make sense to simply do the `BeginInvoke()` for the user?